### PR TITLE
Pdf link bug

### PR DESCRIPTION
**Pull request by denlight**

### CODE REVIEW

File: src/common/services/api/donations.service.js [hunk 1]
```diff
@@ -74,12 +74,12 @@ function DonationsService( cortexApiService, profileService, commonService ) {
...
       .map( ( response ) => {
-        angular.forEach( response['receipt-summaries'], ( item ) => {
-          let link = find( response.links, ( r ) => {
-            return r.uri.indexOf( item['transaction-number'] ) != -1;
-          } );
-          item['pdf-link'] = link;
-        } );
+        let links = filter(response.links, (link) => {
+          return link.rel == 'element';
+        }) || [];
```
Comment (OzzieOrca) [inline src/common/services/api/donations.service.js]: I think filter always returns an array. I tried `_.filter(undefined, () => false)` in the console on the lodash docs page and it returns `[]`.

File: src/common/services/api/donations.service.js [hunk 2]
```diff
@@ -74,12 +74,12 @@ function DonationsService( cortexApiService, profileService, commonService ) {
         data:           data
       } )
       .map( ( response ) => {
```
Comment (OzzieOrca) [inline src/common/services/api/donations.service.js]: This seems to be missing tests.

File: src/common/services/api/donations.service.js [hunk 3]
```diff
@@ -74,12 +74,12 @@ function DonationsService( cortexApiService, profileService, commonService ) {
...
-          let link = find( response.links, ( r ) => {
-            return r.uri.indexOf( item['transaction-number'] ) != -1;
-          } );
-          item['pdf-link'] = link;
-        } );
+        let links = filter(response.links, (link) => {
+          return link.rel == 'element';
+        }) || [];
+        for(let i=0; i < links.length; i++) {
+          response['receipt-summaries'][i]['pdf-link'] = links[i];
```
Comment (OzzieOrca) [inline src/common/services/api/donations.service.js]: Are we sure that the link elements are in the same order as the receipt-summaries? There's not a whole lot you can do on your end. Just want to make sure.

File: src/common/services/api/donations.service.js (hunk 1 above)
Comment (denlight) [inline src/common/services/api/donations.service.js]: Yeah, good catch. I just changed it

File: src/common/services/api/donations.service.js (hunk 3 above)
Comment (denlight) [inline src/common/services/api/donations.service.js]: Apparently yes. BE team promises that it will always be in a right order.

Comment (OzzieOrca) [inline src/common/services/api/donations.service.js]: K cool.

File: src/common/services/api/donations.service.js (hunk 2 above)
Comment (denlight) [inline src/common/services/api/donations.service.js]: @OzzieOrca , tests added

**Pull request merged**